src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -1008,9 +1008,21 @@ def load_arguments(self, _):
         with self.argument_context(scope) as c:
             arg_group = 'Managed Service Identity' if scope.split()[-1] == 'create' else None
             c.argument('identity_scope', options_list=['--scope'], arg_group=arg_group, help="Scope that the system assigned identity can access")
-            c.argument('identity_role', options_list=['--role'], arg_group=arg_group, help="Role name or id the system assigned identity will have")
             c.ignore('identity_role_id')
 
+    for scope in ['vm create', 'vmss create']:
+        with self.argument_context(scope) as c:
+            arg_group = 'Managed Service Identity' if scope.split()[-1] == 'create' else None
```

Comment:
`scope.split()[-1] == 'create'` will always be `True`, right? 😉

Reply:
Good catch! updated